crates/re_viewport/src/space_view.rs

```diff
@@ -84,6 +73,47 @@ impl SpaceViewBlueprint {
         }
     }
 
+    /// Placeholder name displayed in the UI if the user hasn't explicitly named the space view.
+    #[allow(clippy::unused_self)]
+    pub fn missing_name_placeholder(&self) -> String {
+        let entity_path = self
+            .space_origin
+            .as_slice()
+            .iter()
+            .rev()
+            .fold_while(String::new(), |acc, path| {
+                if acc.len() > 10 {
+                    FoldWhile::Done(format!(".../{}", acc))
+                } else {
+                    FoldWhile::Continue(format!(
+                        "{}{}{}",
+                        path.ui_string(),
+                        if acc.is_empty() { "" } else { "/" },
+                        acc
+                    ))
+                }
+            })
+            .into_inner();
+
+        if entity_path.is_empty() {
+            "/".to_owned()
+        } else {
+            entity_path
+        }
+    }
+
+    /// Returns this space view's display name, along with a flag indicating whether it has actually been set by the
+    /// user or not.
+    ///
+    /// When the flag is `false`, the UI should display the resulting name in italics and with a gamma of 0.5 over the
+    /// text color.
```

Comment:
here it would be nice to return an `enum`, e.g. `enum Name { UserChosen, Generated }`.

Named booleans is such a source of confusion and bugs imho